src/hotspot/share/opto/loopnode.cpp

```diff
@@ -3566,26 +3571,48 @@ void PhaseIdealLoop::collect_potentially_useful_predicates(
   }
 }
 
+// Get all skeleton predicate Opaque4 nodes and put them on the 'predicates' list.
+void PhaseIdealLoop::get_skeleton_predicate_opaque_nodes(Node* entry, Unique_Node_List& predicates) {
+  Unique_Node_List list;
+  get_skeleton_predicates(entry, list); // List of projections.
+  // Fetch the Opaque4 nodes and put them on the predicates list.
+  for (uint i = 0; i < list.size(); i++) {
+    Node* opaq = list.at(i)->in(0)->in(1);
+    assert(opaq->Opcode() == Op_Opaque4, "must be Opaque4");
+    predicates.push(opaq);
+  }
+}
+
```

Comment:
I didn't realize you would have to go through an extra list. So I guess what you had before with the get_opaque parameter was better.

Reply:
Yes, we need an extra one. I reverted that part back again.